### ISSUE TITLE
crs is lost when calling representative_point()

### ISSUE THREAD

**Issue by galak75** [geopandas/geopandas#1068]
calling `reprensetative_point()` method on a `GeoSeries` returns a `GeoSeries` without any `crs`:

``` python
from shapely.geometry import Polygon
import geopandas as gpd

square = Polygon(((0, 0), (0, 1), (1, 1), (1, 0), (0, 0)))
gdf = gpd.GeoDataFrame(geometry=[square], crs={'init': 'epsg:32188'})

print(gdf.crs)
# {'init': 'epsg:32188'}

print(gdf.geometry.crs)
# {'init': 'epsg:32188'}

print(gdf.geometry.centroid.crs)
# {'init': 'epsg:32188'}

print(gdf.geometry.representative_point().crs)
# None
```

**Comment by martinfleis** [comment]
Thanks for the report. Honestly, I have no idea why is that happening as it uses exactly the same function (`_delegate_geo_method`) as `buffer` or `simplify` and they return a crs. I can't figure it out at the moment, so if anyone is willing to dig into it, please do so.

**Comment by bahaugen** [comment]
What version of the library are you using? I ran the same commands using v0.5.1 and I couldn't reproduce the error.

```
from shapely.geometry import Polygon
import geopandas as gpd

square = Polygon(((0, 0), (0, 1), (1, 1), (1, 0), (0, 0)))
gdf = gpd.GeoDataFrame(geometry=[square], crs={'init': 'epsg:32188'})

print(gdf.crs)
# {'init': 'epsg:32188'}

print(gdf.geometry.crs)
# {'init': 'epsg:32188'}

print(gdf.geometry.centroid.crs)
# {'init': 'epsg:32188'}

print(gdf.geometry.representative_point().crs)
# {'init': 'epsg:32188'}

print(gdf.geometry.representative_point().crs)
# {'init': 'epsg:32188'}
```

**Comment by martinfleis** [comment]
@bahaugen can you post your `conda list`? I have 0.5.1 as well and I can reproduce it. With 0.5.1, 0.5.0 even 0.4.0. I would like to know what could cause it, as version of geopandas is clearly not.

**Comment by bahaugen** [comment]
Hmm, that is interesting. Here is the conda list of my environment.

```
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main  
attrs                     19.1.0                     py_0    conda-forge
boost-cpp                 1.68.0            h11c811c_1000    conda-forge
bzip2                     1.0.8                h516909a_0    conda-forge
ca-certificates           2019.6.16            hecc5488_0    conda-forge
cairo                     1.16.0            h18b612c_1001    conda-forge
certifi                   2019.6.16                py37_1    conda-forge
cfitsio                   3.470                hb60a0a2_1    conda-forge
click                     7.0                        py_0    conda-forge
click-plugins             1.1.1                      py_0    conda-forge
cligj                     0.5.0                      py_0    conda-forge
curl                      7.65.3               hf8cf82a_0    conda-forge
expat                     2.2.5             he1b5a44_1003    conda-forge
fiona                     1.8.6            py37h5f563d9_4    conda-forge
fontconfig                2.13.1            he4413a7_1000    conda-forge
freetype                  2.10.0               he983fc9_0    conda-forge
freexl                    1.0.5             h14c3975_1002    conda-forge
gdal                      2.4.2            py37h5f563d9_5    conda-forge
geopandas                 0.5.1                      py_0    conda-forge
geos                      3.7.2                he1b5a44_1    conda-forge
geotiff                   1.5.1                h560c3f3_2    conda-forge
gettext                   0.19.8.1          hc5be6a0_1002    conda-forge
giflib                    5.1.7                h516909a_1    conda-forge
glib                      2.58.3            h6f030ca_1002    conda-forge
hdf4                      4.2.13            h9a582f1_1002    conda-forge
hdf5                      1.10.5          nompi_h3c11f04_1100    conda-forge
icu                       58.2              hf484d3e_1000    conda-forge
jpeg                      9c                h14c3975_1001    conda-forge
json-c                    0.13.1            h14c3975_1001    conda-forge
kealib                    1.4.10            h58c409b_1005    conda-forge
krb5                      1.16.3            h05b26f9_1001    conda-forge
libblas                   3.8.0               10_openblas    conda-forge
libcblas                  3.8.0               10_openblas    conda-forge
libcurl                   7.65.3               hda55be3_0    conda-forge
libdap4                   3.20.2            hd48c02d_1000    conda-forge
libedit                   3.1.20170329      hf8c457e_1001    conda-forge
libffi                    3.2.1             he1b5a44_1006    conda-forge
libgcc-ng                 9.1.0                hdf63c60_0  
libgdal                   2.4.2                h524de44_5    conda-forge
libgfortran-ng            7.3.0                hdf63c60_0  
libiconv                  1.15              h516909a_1005    conda-forge
libkml                    1.3.0             h328b03d_1009    conda-forge
liblapack                 3.8.0               10_openblas    conda-forge
libnetcdf                 4.6.2             h056eaf5_1002    conda-forge
libopenblas               0.3.6                h6e990d7_5    conda-forge
libpng                    1.6.37               hed695b0_0    conda-forge
libpq                     11.4                 hd9ab2ff_3    conda-forge
libspatialindex           1.9.0                he1b5a44_1    conda-forge
libspatialite             4.3.0a            he1bb1e1_1029    conda-forge
libssh2                   1.8.2                h22169c7_2    conda-forge
libstdcxx-ng              9.1.0                hdf63c60_0  
libtiff                   4.0.10            h648cc4a_1001    conda-forge
libuuid                   2.32.1            h14c3975_1000    conda-forge
libxcb                    1.13              h14c3975_1002    conda-forge
libxml2                   2.9.9                h13577e0_2    conda-forge
lz4-c                     1.8.3             he1b5a44_1001    conda-forge
munch                     2.3.2                      py_0    conda-forge
ncurses                   6.1               hf484d3e_1002    conda-forge
numpy                     1.16.4           py37h95a1406_0    conda-forge
openblas                  0.3.6                h6e990d7_5    conda-forge
openjpeg                  2.3.1                h58a6597_0    conda-forge
openssl                   1.1.1c               h516909a_0    conda-forge
pandas                    0.25.0           py37hb3f55d8_0    conda-forge
pcre                      8.41              hf484d3e_1003    conda-forge
pip                       19.2.1                   py37_0    conda-forge
pixman                    0.38.0            h516909a_1003    conda-forge
poppler                   0.67.0               ha967d66_7    conda-forge
poppler-data              0.4.9                         1    conda-forge
postgresql                11.4                 hc63931a_3    conda-forge
proj4                     6.1.0                he751ad9_2    conda-forge
pthread-stubs             0.4               h14c3975_1001    conda-forge
pyproj                    2.2.1            py37hc44880f_0    conda-forge
python                    3.7.3                h33d41f4_1    conda-forge
python-dateutil           2.8.0                      py_0    conda-forge
pytz                      2019.1                     py_0    conda-forge
readline                  8.0                  hf8c457e_0    conda-forge
rtree                     0.8.3           py37h666c49c_1002    conda-forge
setuptools                41.0.1                   py37_0    conda-forge
shapely                   1.6.4           py37hec07ddf_1006    conda-forge
six                       1.12.0                py37_1000    conda-forge
sqlite                    3.29.0               hcee41ef_0    conda-forge
tk                        8.6.9             hed695b0_1002    conda-forge
tzcode                    2019a             h516909a_1002    conda-forge
wheel                     0.33.4                   py37_0    conda-forge
xerces-c                  3.2.2             hea5cb30_1003    conda-forge
xorg-kbproto              1.0.7             h14c3975_1002    conda-forge
xorg-libice               1.0.10               h516909a_0    conda-forge
xorg-libsm                1.2.3             h84519dc_1000    conda-forge
xorg-libx11               1.6.8                h516909a_0    conda-forge
xorg-libxau               1.0.9                h14c3975_0    conda-forge
xorg-libxdmcp             1.1.3                h516909a_0    conda-forge
xorg-libxext              1.3.4                h516909a_0    conda-forge
xorg-libxrender           0.9.10            h516909a_1002    conda-forge
xorg-renderproto          0.11.1            h14c3975_1002    conda-forge
xorg-xextproto            7.3.0             h14c3975_1002    conda-forge
xorg-xproto               7.0.31            h14c3975_1007    conda-forge
xz                        5.2.4             h14c3975_1001    conda-forge
zlib                      1.2.11            h516909a_1005    conda-forge
zstd                      1.4.0                h3b9ef0a_0    conda-forge
```

**Comment by martinfleis** [comment]
Thanks. However, this is beyond my knowledge. I can replicate the behaviour in the Conda env generated this way  `conda create -n gp -c conda-forge -c defaults --override-channels python=3 geopandas` on all macOS, Linux and Windows. And there is basically no difference between environment generated by this command and the one you listed above. I have no clue. `buffer` works, using the exactly the same code to get crs...

**Comment by bahaugen** [comment]
Short Version: I think this is fixed on the master branch but not on the 0.5.x branch

Long Version:
When I was running the code above I was in the directory of my local copy of the geopandas repo. When I imported geopandas it was checking my current directory for libraries and found the master branch of the geopandas repo. I think the relevant code was changed in #993 which has been merged into master but it doesn't look like it was added to the 0.5.1 release. Sorry for any confusion I added to the situation.

**Comment by martinfleis** [comment]
Confirmed. Apology not needed, you actually resolved the issue! Thank you!

I confused myself as well by looking at the code on master and not seeing the reason for this behaviour. 

Thank you @galak75 for reporting and @bahaugen for resolving. Closing now as it is already fixed in master by #933.

**Comment by galak75** [comment]
Very good news!!
@martinfleis, @bahaugen : Thank you a lot for digging into it !
:1st_place_medal:
